### PR TITLE
closes #121 automated notification emails

### LINKED ISSUE #121: Email Notifications
When bidding closes on a job,
And a lister accepts a bid,
Emails are sent to all users who bid on the job (one affirmative, rest rejections).

When a lister marks a job as complete,
An email is sent to the user who performed the job prompting for feedback.

### PR DESCRIPTION
**Pull request by jwperry**
Closes #121 
- Automatic acceptance and rejection emails to users upon job bidding end.
- Automatic emails to prompt for feedback once lister marks job as complete.
- Testing for email functionality

### CODE REVIEW

File: app/mailers/notification_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class NotificationMailer < ApplicationMailer
+  def notify_bidder(bid)
+    @bid = bid
+    mail(to: bid.user.email_address, subject: notify_subject_line(bid))
+  end
+
+  def notify_subject_line(bid)
+    if bid.status == "accepted"
```
Comment (ToniRib) [inline app/mailers/notification_mailer.rb]: I think we could clean this up by using `bid.accepted?`

File: app/mailers/notification_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      "Your bid for #{bid.job.title} has been accepted!"
+    else
+      "Your bid for #{bid.job.title} has not succeeded."
+    end
+  end
+
+  def feedback_prompt(job)
+    @job = job
+    contractor_email = job.bids.find_by(status: 1).user.email_address
```
Comment (ToniRib) [inline app/mailers/notification_mailer.rb]: You can also use `job.contractor_for_selected_bid` instead of `job.bids.find_by(status: 1).user`

**Comment by ToniRib** [comment]
Hey JP I refactored a little to use some more of the helper methods that we have just to make the code a little cleaner. Otherwise this looks good! Thanks for doing this!

File: app/views/notification_mailer/feedback_prompt.html.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+The job <%= "#{@job.title}" %> has been marked as complete by the lister. Well done!
```
Comment (danjwinter) [inline app/views/notification_mailer/feedback_prompt.html.erb:1]: so so jolly!

**Comment by ToniRib** [comment]
Looks good, I'm going to try this out in Heroku.

**Pull request merged**